ucm2/sof-soundwire/HiFi.conf

```diff
@@ -28,4 +28,13 @@ If.hsdev {
 	False.Include.hsdev.File "/sof-soundwire/${var:HeadsetCodec1}.conf"
 }
 
+If.dmicdev {
```

Comment:
This should be merged to `If.micdev`. I assume that there will be never mix of SDW and direct DMIC.

Reply:
It's likely indeed, but I don't get what you meant by merging. We use

rt715-> mic:rt715 and if found we use a dedicated init.conf
dmic -> cfg-dmics:1..4

Reply:
I assume that having both 'mic:XXX and cfg-dmic:1+' is invalid, right? So add the second If inside If.micdev something like:

<pre>
If.micdev {
        Condition {
                Type String
                Empty "${var:MicCodec1}"
        }
        False.Include.micdev.File "/sof-soundwire/${var:MicCodec1}.conf"
        True.If {
                Condition {
                        Type String
                        Empty "${var:Dmics1}"
                }
                False.Include.dmic.File "/sof-soundwire/dmic.conf"
        }
}
</pre>

You're true. It's not direct merging.

Reply:
it depends what we want. If mic:xyz is the name of the input, we could have mic:rt715 or mic:dmic.
cfg-dmics was added specifically for dmic but maybe it should be cfg-mic in case we want to deal with other configurations (e.g. for cfg-spk we don't care what type of amp is used).

Reply:
My configuration works with both `mic:dmic` and `cfg-dmic:2` option without `mic:dmic`. If both are passed by the driver, then the second if is not required (dmic will be extracted from `mic:dmic`, right?).

But with `mic:rt715 cfg-dmics:4` we are in trouble, but I assume that it's invalid hw combo (direct dmics + SDW dmics), right?